src/visitors/sympy_replace_solutions_visitor.cpp

```diff
@@ -0,0 +1,350 @@
+/*************************************************************************
+ * Copyright (C) 2018-2019 Blue Brain Project
+ *
+ * This file is part of NMODL distributed under the terms of the GNU
+ * Lesser General Public License. See top-level LICENSE file for details.
+ *************************************************************************/
+
+#include "visitors/sympy_replace_solutions_visitor.hpp"
+#include "visitors/lookup_visitor.hpp"
+
+
+#include "ast/all.hpp"
+#include "utils/logger.hpp"
+#include "visitors/visitor_utils.hpp"
+
+namespace nmodl {
+namespace visitor {
+
+
+void SympyReplaceSolutionsVisitor::InterleavesCounter::new_equation(const bool is_in_system) {
+    n_ += (!in_system_ && is_in_system);
+    in_system_ = is_in_system;
+}
+
+SympyReplaceSolutionsVisitor::SympyReplaceSolutionsVisitor(
+    const std::vector<std::string>& pre_solve_statements,
+    const std::vector<std::string>& solutions,
+    const std::unordered_set<ast::Statement*>& to_be_removed,
+    const ReplacePolicy policy,
+    const size_t n_next_equations)
+    : pre_solve_statements_(pre_solve_statements.begin(), pre_solve_statements.end(), 2)
+    , to_be_removed_(&to_be_removed)
+    , policy_(policy)
+    , n_next_equations_(n_next_equations)
+    , replaced_statements_begin_(-1)
+    , replaced_statements_end_(-1) {
+    const auto ss_tmp_delimeter =
+        std::find_if(solutions.begin(), solutions.end(), [](const std::string& statement) {
+            return statement.substr(0, 3) != "tmp";
```

Comment:
I opened an issue that reflects the TODOs that are in the code. I leave it to another PR since it is likely that `tmp*` and `old_*` are hard-coded around the code and a change here can generate bugs